langserver/handlers/complete.go

```diff
@@ -54,10 +54,16 @@ func (h *logHandler) TextDocumentComplete(ctx context.Context, params lsp.Comple
 	p.SetLogger(h.logger)
 	p.SetSchemaReader(sr)
 
+	token, hclPos, err := hclFile.TokenAtPosition(fPos)
```

Comment:
Relatedly we may need to change the interface of `configBlockFactory.New()` to take tokens instead of `hclsyntax.Block` and remove `parser.ParseBlockFromHCL` in favour of `parser.ParseBlockFromTokens(hcl.Tokens)`, so we retain the tokens in the parser, which in turn leads us to a conclusion that the original abstraction of `TokenAtPosition` doesn't work very well here and should probably be reimplemented as a replacement of `BlockAtPosition(fPos) hcl.Block`, e.g. `BlockTokensAtPosition() hclsyntax.Tokens`.

Then finally we can re-implement the original abstraction inside `internal/terraform/lang` as `TokenAtPos(hclsyntax.Tokens, hcl.Pos) hclsyntax.Token`.

Reply:
I will attempt to do the refactoring outside of this PR and see how far I can get.

Reply:
Actually bytes may not be the best format for passing around, since we loose context like position within the whole file. 🤔 I'll try to figure out something with tokens instead.